tests/tool_excimer_profile_test.php

```diff
@@ -95,6 +95,29 @@ public function test_set_flamedata(): void {
         $this->assertEquals($node->value, $profile->get('numsamples'));
     }
 
+    /**
+     * Test to make sure that the parameters value is truncated if it is too long.
+     *
+     * @throws \coding_exception
+     */
+    public function test_set_parameters() {
+        $profile = new profile();
+
+        $exactstring = str_repeat('a', profile::MAX_STR_LEN);
+        $profile->set('parameters', $exactstring);
+        $this->assertEquals($exactstring, $profile->get('parameters'));
+
+        $expectedstring = str_repeat('a', profile::MAX_STR_LEN - 4) . ' ...';
```

Comment:
Will need adjustments as per previous comment, but otherwise the tests seem fine.